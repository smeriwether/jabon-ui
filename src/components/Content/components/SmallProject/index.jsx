import React from 'react';
import PropTypes from 'prop-types';

import ExpandUp from '../../../../UI/ExpandUp';

const propTypes = {
  onExpand: PropTypes.func.isRequired
};

const SmallProject = ({ onExpand }) => {
  return (
    <div className='h-100 w-100 near-black flex items-center justify-between'>
      <div className='b f3 pa2'>
        Project Information
      </div>
      <div className='pointer pa2' data-role='expand' onClick={ onExpand } >
        <ExpandUp />
      </div>
    </div>
  );
};

SmallProject.propTypes = propTypes;

export default SmallProject;
