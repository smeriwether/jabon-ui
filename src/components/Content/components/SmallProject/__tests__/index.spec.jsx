import SmallProject from '../../SmallProject';

describe('<SmallProject />', () => {
  const setup = (overrides) => {
    const props = {
      onExpand: jest.fn(),
      ...overrides
    };

    const wrapper = shallow(<SmallProject { ...props } />);

    return { wrapper };
  };

  it('should match the snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the onExpand prop when the expand button is clicked', () => {
    const onExpandMock = jest.fn();
    const { wrapper } = setup({ onExpand: onExpandMock });

    wrapper.find('div[data-role="expand"]').simulate('click');

    expect(onExpandMock).toHaveBeenCalled();
  });
});
