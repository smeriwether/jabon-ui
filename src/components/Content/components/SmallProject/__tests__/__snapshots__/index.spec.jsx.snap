// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SmallProject /> should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SmallProject
    onExpand={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="b f3 pa2"
        >
          Project Information
        </div>,
        <div
          className="pointer pa2"
          data-role="expand"
          onClick={[MockFunction]}
        >
          <ExpandUp
            fill="#000000"
          />
        </div>,
      ],
      "className": "h-100 w-100 near-black flex items-center justify-between",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Project Information",
          "className": "b f3 pa2",
        },
        "ref": null,
        "rendered": "Project Information",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ExpandUp
            fill="#000000"
          />,
          "className": "pointer pa2",
          "data-role": "expand",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "fill": "#000000",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="b f3 pa2"
          >
            Project Information
          </div>,
          <div
            className="pointer pa2"
            data-role="expand"
            onClick={[MockFunction]}
          >
            <ExpandUp
              fill="#000000"
            />
          </div>,
        ],
        "className": "h-100 w-100 near-black flex items-center justify-between",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Project Information",
            "className": "b f3 pa2",
          },
          "ref": null,
          "rendered": "Project Information",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ExpandUp
              fill="#000000"
            />,
            "className": "pointer pa2",
            "data-role": "expand",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "fill": "#000000",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
